docs/src/components/theme-grid.astro

```diff
@@ -0,0 +1,196 @@
+---
+import { Icon } from '@astrojs/starlight/components';
+import FluidGrid from './fluid-grid.astro';
+import MediaCard from './media-card.astro';
+import ThemeImage from './theme-image.astro';
+
+interface Props {
+	labels: {
+		/** Accessible label for the theme toggle. */
+		toggle: string;
+		/** Accessible label for the dark color scheme variant. */
+		dark: string;
+		/** Accessible label for the light color scheme variant. */
+		light: string;
+	};
+	themes: Array<{
+		/** The name of this theme. */
+		title: string;
+		/** A short description of this theme. */
+		description: string;
+		/** URL for this theme’s website. Ideally should link to a demo site showing off the theme. */
+		href: string;
+		previews: {
+			/** Image filename as found in `src/assets/themes/`, e.g. `ion-light.png`. */
+			light: string;
+			/** Image filename as found in `src/assets/themes/`, e.g. `ion-dark.png`. */
+			dark: string;
+		};
+	}>;
+}
+
+const { labels, themes } = Astro.props;
+---
+
+<FluidGrid minColumnWidth="25rem" gap="clamp(1rem, calc(0.75rem + 1vw), 1.5rem)">
+	{
+		themes.map(async ({ title, description, href, previews }) => {
+			if (!previews) return;
+			const inputName = `preview-${title.toLowerCase().replace(/[^a-z0-9_-]+/g, '')}`;
+			return (
+				<MediaCard>
+					<starlight-theme-preview slot="media">
+						<ThemeImage src={previews.light} class="light" />
+						<ThemeImage src={previews.dark} class="dark" />
+						<fieldset>
+							<legend class="sr-only">{labels.toggle}</legend>
+							<label class="dark">
+								<Icon name="approve-check" />
+								<span class="sr-only">{labels.dark}</span>
+								<input class="sr-only dark" name={inputName} value="dark" type="radio" checked />
+							</label>
+							<label class="light">
+								<Icon name="approve-check" />
+								<span class="sr-only">{labels.light}</span>
+								<input class="sr-only light" name={inputName} value="light" type="radio" />
+							</label>
+						</fieldset>
```

Comment:
Considering some [well-known](https://bugs.webkit.org/show_bug.cgi?id=230824) screen reader issues with fieldsets, e.g. repeating everything twice, or even three times for me for the legend in Firefox, that's a lot of content for screen reader to go through before even getting to the theme name. I wonder if markup wise they should be inverted and swap only visually.

If compared to the video content showcase, these cards would just announce "Play" then the video title which is quite small in this case.